### PR TITLE
Specifying gRPC event engine

### PR DESCRIPTION
**Pull request by overvenus**
Set gRPC event engine to the epollsig.

screenshot from 2018-01-09 14-32-51

### CODE REVIEW

**Comment by siddontang** [comment]
why not using epollex?

**Comment by siddontang** [comment]
how about the performance result in BenchBot?

File: src/bin/tikv-server.rs [hunk 1]
```diff
@@ -494,6 +494,11 @@ fn main() {
     // Before any startup, check system configuration.
     check_system_config(&config);
 
+    // Set gRPC event engine to epollsig.
+    // See more: https://github.com/grpc/grpc/blob/486761d04e03a9183d8013eddd86c3134d52d459\
+    //           /src/core/lib/iomgr/ev_posix.cc#L149
+    env::set_var("GRPC_POLL_STRATEGY", "epollsig");
```
Comment (siddontang) [inline src/bin/tikv-server.rs]: do we need to check GRPC_POLL_STRATEGY env at first, so maybe we can use another poll directly for the benchmark.

File: src/bin/tikv-server.rs [hunk 2]
```diff
@@ -330,6 +330,19 @@ fn overwrite_config_with_cmd_args(config: &mut TiKvConfig, matches: &ArgMatches)
...
 
+// Set gRPC event engine to epollsig.
+// See more: https://github.com/grpc/grpc/blob/486761d04e03a9183d8013eddd86c3134d52d459\
+//           /src/core/lib/iomgr/ev_posix.cc#L149
+fn specify_grpc_event_engine() {
+    const GRPC_POLL_STRATEGY: &str = "GRPC_POLL_STRATEGY";
+    let mut engines = "epollsig".to_owned();
+    if let Ok(env_engines) = env::var(GRPC_POLL_STRATEGY) {
+        engines.push(',');
+        engines.push_str(&env_engines);
```
Comment (siddontang) [inline src/bin/tikv-server.rs]: so we will still use epollsig even setting another poll in env?

File: src/bin/tikv-server.rs [hunk 3]
```diff
@@ -330,6 +330,18 @@ fn overwrite_config_with_cmd_args(config: &mut TiKvConfig, matches: &ArgMatches)
     }
 }
 
+// Set gRPC event engine to epollsig.
+// See more: https://github.com/grpc/grpc/blob/486761d04e03a9183d8013eddd86c3134d52d459\
+//           /src/core/lib/iomgr/ev_posix.cc#L149
+fn specify_grpc_event_engine() {
```
Comment (BusyJay) [inline src/bin/tikv-server.rs]: s/specify/configure/

File: src/bin/tikv-server.rs [hunk 4]
```diff
@@ -330,6 +330,18 @@ fn overwrite_config_with_cmd_args(config: &mut TiKvConfig, matches: &ArgMatches)
     }
 }
 
+// Set gRPC event engine to epollsig.
+// See more: https://github.com/grpc/grpc/blob/486761d04e03a9183d8013eddd86c3134d52d459\
+//           /src/core/lib/iomgr/ev_posix.cc#L149
+fn configure_grpc_event_engine() {
```
Comment (siddontang) [inline src/bin/tikv-server.rs]: why noting using grpc_poll_strategy directly?

**Comment by BusyJay** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Comment by july2993** [comment]
does Tikv only supposed to be run on linux?  i run on my mac previously, this PR will make you must to specify the GRPC_POLL_STRATEGY env in non linux platform, or it will fail to set up using the epoll engine if i am not wrong, better to use default engine in non linux platform

**Comment by siddontang** [comment]
oh, @overvenus please fix it.

**Comment by overvenus** [comment]
Hi @july2993 It's now fixed. Thank you for your feedback!

**Pull request merged**